CharlieBackend.Api.UnitTes/LessonServiceTests.cs

```diff
@@ -0,0 +1,404 @@
+                   Mentor = mentor,
+                   StudentGroup = studentGroup,
+                   Theme = theme,
+                   Visits = default
+                }
+            };
+
+            List<LessonDto> lessonsDto = new List<LessonDto>
+            {
+                new LessonDto
+                {
+                   Id = 7,
+                   ThemeName = "ExampleName",
+                   MentorId = 2,
+                   StudentGroupId = 3,
+                   LessonDate = DateTime.Parse("2020-11-18T15:00:00.384Z"),
+                   LessonVisits = default
+                },
+                new LessonDto
+                {
+                   Id = 8,
+                   ThemeName = "ExampleName",
+                   MentorId = 2,
+                   StudentGroupId = 3,
+                   LessonDate = DateTime.Parse("2020-11-18T15:00:00.384Z"),
+                   LessonVisits = default
+                },
+                new LessonDto
+                {
+                   Id = 9,
+                   ThemeName = "ExampleName",
+                   MentorId = 2,
+                   StudentGroupId = 3,
+                   LessonDate = DateTime.Parse("2020-11-18T15:00:00.384Z"),
+                   LessonVisits = default
+                }
+            };
+
+            var lessonsRepositoryMock = new Mock<ILessonRepository>();
+            lessonsRepositoryMock.Setup(x => x.GetAllAsync())
+                .ReturnsAsync(lessons);
+
+            _unitOfWorkMock.Setup(x => x.LessonRepository).Returns(lessonsRepositoryMock.Object);
+
+            var lessonService = new LessonService(
+               _unitOfWorkMock.Object,
+               _mapper
+               );
+
+            //Act
+            var result = await lessonService.GetAllLessonsAsync();
+
+            //Assert
+            Assert.NotNull(result);
+
+            Assert.Equal(lessonsDto.Count, result.Count);
+
+            for (int i = 0; i < result?.Count; i++)
+            {
+                Assert.Equal(lessonsDto[i].Id, result[i].Id);
+                Assert.Equal(lessonsDto[i].LessonDate, result[i].LessonDate);
+
+                for (int j = 0; j < result[i].LessonVisits?.Count; j++)
+                {
+                    Assert.Equal(lessonsDto[i].LessonVisits[j]?.Comment, result[i].LessonVisits[j]?.Comment);
+                    Assert.Equal(lessonsDto[i].LessonVisits[j]?.Presence, result[i].LessonVisits[j]?.Presence);
+                    Assert.Equal(lessonsDto[i].LessonVisits[j]?.StudentId, result[i].LessonVisits[j]?.StudentId);
+                    Assert.Equal(lessonsDto[i].LessonVisits[j]?.StudentMark, result[i].LessonVisits[j]?.StudentMark);
+                }
+                
+                Assert.Equal(lessonsDto[i].MentorId, result[i].MentorId);
+                Assert.Equal(lessonsDto[i].StudentGroupId, result[i].StudentGroupId);
+                Assert.Equal(lessonsDto[i].ThemeName, result[i].ThemeName);
+            }
+        }
```

Comment:
It's necessary? You mock the return GetAll result of the repository, but the method GetAllLessonsAsync of the Lesson service has no other data processing logic, so if you are mock repository GetAll method test is useless

Reply:
Is it better to remove fully the test for GetAllLessons() ?

Reply:
Yeap